stdpopsim/slim_engine.py

```diff
@@ -948,17 +948,54 @@ def fix_time(event):
     condition_on_allele_frequency = []
     op_id = stdpopsim.ext.ConditionOnAlleleFrequency.op_id
     slim_mutation_ids = []
-    for x in _dfe_to_mtypes(contig).values():
+    slim_mutation_dfe = []
+    for i, x in _dfe_to_mtypes(contig).items():
         slim_mutation_ids.extend([u[0] for u in x])
+        slim_mutation_dfe.extend([i for u in x])
     for ee in extended_events:
+        if hasattr(ee, "mutation_id"):
+            mut_id = getattr(ee, "mutation_id")
+            cls_name = ee.__class__.__name__
+            mtype_idx = [
+                j
+                for j, i in enumerate(slim_mutation_dfe)
+                if contig.dfe_list[i].id == mut_id
+            ]
+            n_mtypes = len(mtype_idx)
+            # TODO: are these checks too restrictive? For example, these will
+            # prevent using DrawMutation with an arbitrary MutationType in an
```

Comment:
Yup, that should be possible, because the check is just ensuring that the mutation ID in each extended event maps to a single mutation (so the DFE id is synonymous with the mutation id in SLiM). These have no associated intervals, so should not "intersect" with any other DFEs. So, these point mutations can coexist with an arbitrary number of other DFEs with multiple mutation types and non-unique ids.